Mathlib/Dynamics/TopologicalEntropy/NetEntropy.lean

```diff
@@ -0,0 +1,408 @@
+    (n : ℕ) :
+    netMaxcard T F U n ≤ coverMincard T F U n := by
+  rcases eq_top_or_lt_top (coverMincard T F U n) with h' | h'
+  · exact h' ▸ le_top
+  · rcases ((coverMincard_finite_iff T F U n).1 h') with ⟨t, t_cover, t_mincard⟩
+    rw [← t_mincard]
+    exact iSup₂_le (fun s s_net ↦ Nat.cast_le.2 (s_net.card_le_card_of_isDynCoverOf h t_cover))
+
+/-- Given an entourage `U` and a time `n`, a minimal dynamical cover by `U ○ U` has a smaller
+  cardinal than a maximal dynamical net by `U`. This lemma is the second of two key results to
+  compare two versions topological entropy: with cover and with nets.-/
+lemma coverMincard_le_netMaxcard (T : X → X) (F : Set X) {U : Set (X × X)} (U_rfl : idRel ⊆ U)
+    (U_symm : SymmetricRel U) (n : ℕ) :
+    coverMincard T F (U ○ U) n ≤ netMaxcard T F U n := by
+  classical
+  -- WLOG, there exists a maximal dynamical net `s`.
+  rcases (eq_top_or_lt_top (netMaxcard T F U n)) with h | h
+  · exact h ▸ le_top
+  rcases ((netMaxcard_finite_iff T F U n).1 h) with ⟨s, s_net, s_netMaxcard⟩
+  rw [← s_netMaxcard]
+  apply IsDynCoverOf.coverMincard_le_card
+  --  We have to check that `s` is a cover for `dynEntourage T F (U ○ U) n`.
+  -- If `s` is not a cover, then we can add to `s` a point `x` which is not covered
+  -- and get a new net. This contradicts the maximality of `s`.
+  by_contra h
+  rcases not_subset.1 h with ⟨x, x_F, x_uncov⟩
+  simp only [Finset.mem_coe, mem_iUnion, exists_prop, not_exists, not_and] at x_uncov
+  have larger_net : IsDynNetOf T F U n (insert x s) := by
+    apply And.intro (insert_subset x_F s_net.1)
+    refine pairwiseDisjoint_insert.2 (And.intro s_net.2 (fun y y_s _ ↦ ?_))
+    refine disjoint_left.2 (fun z z_x z_y ↦ x_uncov y y_s ?_)
+    exact mem_ball_dynEntourage_comp T n U_symm x y (nonempty_of_mem ⟨z_x, z_y⟩)
+  rw [← Finset.coe_insert x s] at larger_net
+  apply larger_net.card_le_netMaxcard.not_lt
+  rw [← s_netMaxcard, Nat.cast_lt]
+  refine (lt_add_one s.card).trans_eq (Finset.card_insert_of_not_mem fun x_s ↦ ?_).symm
+  apply x_uncov x x_s
+  apply ball_mono (dynEntourage_monotone T n (subset_comp_self U_rfl)) x
+  apply ball_mono (idRel_subset_dynEntourage T U_rfl n) x
+  simp only [ball, mem_preimage, mem_idRel]
+
+open ENNReal EReal
+
+lemma log_netMaxcard_nonneg (T : X → X) {F : Set X} (h : F.Nonempty) (U : Set (X × X)) (n : ℕ) :
+    0 ≤ log (netMaxcard T F U n) := by
+  apply zero_le_log_iff.2
+  rw [← ENat.toENNReal_one, ENat.toENNReal_le]
+  exact (one_le_netMaxcard_iff T F U n).2 h
+
+/-! ### Net entropy of entourages -/
+
+open Filter
+
+/-- The entropy of an entourage `U`, defined as the exponential rate of growth of the size of the
+largest `(U, n)`-dynamical net of `F`. Takes values in the space of extended real numbers
+`[-∞,+∞]`. This version uses a `limsup`, and is chosen as the default definition.-/
+noncomputable def netEntropyEnt (T : X → X) (F : Set X) (U : Set (X × X)) :=
```

Comment:
I'm just realizing that the `Ent` for entourage collides with the `Ent` for entropy. Given this, you should use longer names, like `netEntropyEntour` or `netEntropyEntourage`. And do the same in the other file...

Reply:
Done in both files. I've chosen `netEntropyEntourage`, because there is not much to gain by deleting 3 letters of an already long name.